daisy/instance.go

```diff
@@ -322,15 +326,19 @@ func (i *InstanceBeta) setSourceMachineImage(machineImage string) {
 func (i *InstanceBeta) register(name string, s *Step, ir *instanceRegistry, errs DError) {
 	// Register disk attachments.
 	for _, d := range i.Disks {
-		dName := d.Source
+		diskName := d.Source
 		if d.InitializeParams != nil {
 			parts := NamedSubexp(diskTypeURLRgx, d.InitializeParams.DiskType)
 			if parts["disktype"] == "local-ssd" {
 				continue
 			}
-			dName = d.InitializeParams.DiskName
+			diskName = d.InitializeParams.DiskName
+		}
+		deviceName := d.DeviceName
+		if deviceName == "" {
+			deviceName = diskName
 		}
```

Comment:
related to the above comment, moving this to regAttach would avoid code duplication, unless there's a need for deviceName being empty in some scenarios

Reply:
done